### PR TITLE
Transaction pool

### PR DESCRIPTION
**Pull request by nhunzaker**
This PR sets up an object pool for transactions and greatly simplifies the transaction creation process.

### CODE REVIEW

File: src/Microcosm.js [hunk 1]
```diff
@@ -39,68 +41,43 @@ let Microcosm = function() {
...
-    this.transactions.splice(this.transactions.indexOf(transaction), 1)
-  },
-
-  transactionDidFail(transaction) {
-    this.release(transaction)
-    return this.rollforward()
+    if (!complete && has(this.transactions, transaction) === false) {
+      this.transactions = this.transactions.concat(transaction).sort(sortById)
+    }
   },
```
Comment (nhunzaker) [inline src/Microcosm.js]: Actually, I always want to push these in to maintain history for those who want it.

File: src/Transaction.js [hunk 2]
```diff
@@ -4,39 +4,29 @@
...
-  return {
-    type: `${ type }`,
-    payload,
-    meta: {
-      active: false,
-      done: false
-    },
-    error: false
-  }
+const blueprint = {
```
Comment (solomonhawk) [inline src/Transaction.js]: Capitalize const bindings?

Comment (nhunzaker) [inline src/Transaction.js]: Yes

File: src/lifecycle.js [hunk 3]
```diff
@@ -4,13 +4,15 @@
  * actions.
  */
 
-exports.mapping = {
+const mapping = {
```
Comment (solomonhawk) [inline src/lifecycle.js]: Uppercase

**Comment by solomonhawk** [comment]
This looks really nice! :+1: Can you give an example of using the transaction lifecycle hooks?

**Comment by nhunzaker** [comment]
@solomonhawk The plan is to eventually expose those to plugins, but for stores, the lifecycle hooks are actions, so you can register to them like:


```
import { willStart } from 'microcosm/lifecycle'

let Store = {
    reset() {
        return []
    },
    register() {
        return {
            [willStart]: Store.reset
        }
    }    

}
```

Or, of course, you could just use `Store.getInitialState`.  The long term plan is to expose more hooks, but I wanted to get this transaction stuff clamped down first.

**Comment by nhunzaker** [comment]
Basically, a method that will call the same hooks on Plugins. Soon to come! I believe they will be something like:

```
transactionDidOpen
transactionDidUpdate
transactionDidClose
transactionDidFail
```

**Comment by nhunzaker** [comment]
(then I can finally close #28 )

**Pull request merged**